File: C/LinkedList/src/linkedlist.c
#include "../include/linkedlist.h"

void initList(struct LinkedList *list ){
    list->head = NULL ;
}

void insert(struct LinkedList *list, int item)
{
    struct Node *newNode = malloc(sizeof(struct Node));
    newNode->item = item;
    newNode->ptrnext = NULL;        
    if (list->head == NULL)
    {
        list->head = newNode ;
    }
    else 
    {
        struct Node *ptrcurrent = list->head ;
        while (ptrcurrent->ptrnext != NULL)
        {
            ptrcurrent = ptrcurrent->ptrnext ;
        }
        ptrcurrent->ptrnext = newNode ;
        
    }
}
void search(struct LinkedList *list,int item){
    struct Node *ptrcurrent = list->head ;
    while (ptrcurrent!=NULL)
    {
        if (ptrcurrent->item == item)
        {
            printf("Item found: %d\n", item);
            return;
        }
        ptrcurrent= ptrcurrent->ptrnext;
    }
    printf("Item not found: %d\n", item);
}

void delete(struct LinkedList *list,int item){
    struct Node *ptrcurrent = list->head ;
    struct Node *ptrprevious = NULL ;
    while (ptrcurrent!=NULL)
    {
        if (ptrcurrent->item == item)
        {
            if (ptrprevious == NULL)
            {
                list->head = ptrcurrent->ptrnext ;
            }
            else {
                ptrprevious->ptrnext = ptrcurrent->ptrnext ;
            }
            free(ptrcurrent);
            printf("Item deleted: %d\n", item);
            return;
        }
        ptrprevious = ptrcurrent ;
        ptrcurrent= ptrcurrent->ptrnext;
    }
    printf("Item not found for deletion: %d\n", item);
}

void display(struct LinkedList *list)
{
    struct Node *ptrcurrent = list->head ;
    while (ptrcurrent != NULL)
    {
        printf("%d -> ", ptrcurrent->item);
        ptrcurrent = ptrcurrent->ptrnext;
    }
    printf("NULL\n");

}


    

File: python/LinkedList/collection/ds/linkedlist.py
from .node import Node

class LinkedList:

    #constructor
    def __init__(self):
        self.head = None

    # CRUD Operations

    def insert(self, item):
        new_node = Node(item)  

        if self.head is None:
            self.head = new_node
        else:
            current = self.head
            while current.next!= None:
                current = current.next
            current.next = new_node 

    def search(self, item):
        current = self.head
        while current:
            if current.data == item:
                return True
            current = current.next
        return False
    
    def delete(self, item):

        current = self.head
        previous = None
        
        while current and current.data != item:
            previous = current
            current = current.next
        if current is None:
            return  
        if previous is None:
            self.head = current.next  
        else:
            previous.next = current.next
            
    def display(self):
        current = self.head

        while current:
            print(current.data, end=" -> ")
            current = current.next
        print("None")
